Galaxus.Functional/Linq/OptionLinqExtensions.cs

```diff
@@ -123,8 +123,23 @@ public static Option<TSource> SingleOrNone<TSource>(this IEnumerable<TSource> so
                 .Select(Option<TSource>.Some)
                 .SingleOrDefault();
 
+        /// <summary>
+        /// Checks a Dictionary if it contains a value for a given key. If so this value is wraped in a Option. Else <see cref="Option{T}.None"/> is returned.
+        /// </summary>
+        /// <param name="dictionary">The dictionary in quesion</param>
+        /// <param name="key">The key to check if a value exists</param>
+        /// <returns>The Value of the Dictionary or None</returns>
+        public static Option<TSome> GetValueOrNone<TKey, TSome>(
```

Comment:
What does this method have to do with Linq? Wouldn't it make more sense to place it in a class called `OptionCollectionExtensions`? Or maybe even place the class in the namespace `Galaxus.Functional.Collections` and name it `DictionaryExtensions`? This goes well with this issue, imho: #25

Reply:
I wouldn't mind it being with the other LINQ extensions, especially since it is now an extension to `IEnumerable<KeyValuePair<,>>`. Theres not that many extensions, so I don't think we need to spread them out more just yet.

Reply:
I would propose to discuss the architecture of the namespaces in an other issue and if needed, move this method after that discussion.